flow/project.py

```diff
@@ -2077,75 +2156,99 @@ def _get_operations_status(self, jobs, cached_status):
+                        "eligible": eligible,
+                        "completed": completed,
+                    }
 
         yield from sorted(status_dict.items())
 
-    def get_job_status(self, job, ignore_errors=False, cached_status=None):
-        """Return status information about a job.
+    def _get_aggregate_status(self, aggregate, cached_status, ignore_errors=False):
+        """Return status information about an aggregate.
 
         Parameters
         ----------
-        job : :class:`~signac.contrib.job.Job`
-            The signac job.
-        ignore_errors : bool
-            Whether to ignore exceptions raised during status check. (Default value = False)
+        aggregate : tuple of :class:`~signac.contrib.job.Job`
+            Aggregate for which status information is fetched.
         cached_status : dict
             Dictionary of cached status information. The keys are uniquely
             generated ids for each group and job. The values are instances of
             :class:`~.JobStatus`. (Default value = None)
+        ignore_errors : bool
+            Whether to ignore exceptions raised during status check. (Default value = False)
 
         Returns
         -------
         dict
             A dictionary containing job status for all jobs.
 
         """
-        # TODO: Add support for aggregates for this method.
-        if cached_status is None:
-            cached_status = self._get_cached_status()
-        result = {}
-        result["job_id"] = str(job)
+        aggregate_id = get_aggregate_id(aggregate)
+        result = {
+            "job_id": aggregate_id,
+            "operations": {},
+            "_operations_error": None,
+        }
         try:
             result["operations"] = dict(
-                self._get_operations_status((job,), cached_status)
+                self._get_aggregate_group_status(aggregate, cached_status)
             )
-            result["_operations_error"] = None
         except Exception as error:
             logger.debug(
-                "Error while getting operations status for job '%s': '%s'.", job, error
+                "Error while getting operations status for job '%s': '%s'.",
+                aggregate_id,
+                error,
             )
             if ignore_errors:
-                result["operations"] = {}
                 result["_operations_error"] = str(error)
             else:
                 raise
-        try:
-            result["labels"] = sorted(set(self.labels(job)))
-            result["_labels_error"] = None
-        except Exception as error:
-            logger.debug(
-                "Error while determining labels for job '%s': '%s'.", job, error
-            )
-            if ignore_errors:
-                result["labels"] = []
-                result["_labels_error"] = str(error)
-            else:
-                raise
+        return result
+
+    def get_job_status(self, job, ignore_errors=False, cached_status=None):
```

Comment:
This function was heavily simplified by creating `_get_aggregate_status` above and using the duplicated code from `_get_job_labels`.